connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/feel/jackson/FeelFunctionDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package io.camunda.connector.runtime.core.feel.jackson;
+
+import com.fasterxml.jackson.databind.BeanProperty;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import io.camunda.connector.runtime.core.feel.FeelEngineWrapper;
+import java.util.function.Function;
+
+public class FeelFunctionDeserializer<IN, OUT> extends AbstractFeelDeserializer<Function<IN, OUT>> {
+
+  private final Class<OUT> outputType;
+
+  public FeelFunctionDeserializer(Class<OUT> outputType, FeelEngineWrapper feelEngineWrapper) {
+    super(feelEngineWrapper);
+    this.outputType = outputType;
+  }
+
+  private final FeelEngineWrapper feelEngineWrapper = new FeelEngineWrapper();
+
+  @Override
+  protected Function<IN, OUT> doDeserialize(String expression) {
+    return (input) -> {
+      Object result = feelEngineWrapper.evaluate(expression, input);
+      return feelEngineWrapper.getObjectMapper().convertValue(result, outputType);
```

Comment:
Curious why this is necessary. Cant we just return the result casted to the type?

Reply:
As discussed synchronously, we need that to do the final type conversion into the `Function` output type.